### ISSUE TITLE
[Select] Focus indicator does not have sufficient contrast on expanded options

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3129]
## Current behavior

Placing focus inside `sl-listbox` with options results in one options background color change to #C8D1DA. This color has contrast ratio of 1.5:1 with white background. It's the only way focused option is highlighted and it may not be sufficient for people with low vision.

Additionally focus ring is being visible on `sl-select-button` for all the time that listbox is expanded and that may confuse user who is expecting focus to move inside the expanded content.

<img width="856" height="213" alt="Image" src="https://github.com/user-attachments/assets/a99c86f7-5a65-4f77-9678-160419fb0c4e" />

## Suggested solution

It's recommended to use focus indicator in the form of blue ring on focused option and additional line indicator (placed on the left edge of element) as it's now done for [Menu](https://storybook.sanomalearning.design/?path=/story/actions-menu--example).

It's also recommended to make focus indicator on `sl-select-button` invisible when user place focus inside listbox with options. Focus should be visible on `sl-select-button` only when listbox is collapsed.

<img width="1006" height="393" alt="Image" src="https://github.com/user-attachments/assets/6d1ba332-ac50-4bef-822b-4dca4c3a28b0" />

### 👤 Your name

Kacper

### 🧢 Your Product/Team

SLDS
